src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -380,18 +393,17 @@ Payment::doApply()
         // Ripple payment with at least one intermediate step and uses
         // transitive balances.
 
-        if (depositPreauth && reqDepositAuth)
+        if (depositPreauth && depositAuth)
```

Comment:
Shouldn't this be reqDepositAuth? Same comment as in Escrow and PayChan. It is difficult to make sure pre-amendment of old/new code produces the same result.

Reply:
It is diveded into 2 check - one of them is common to all transferring transactions(and it is present in `verifyDepositPreauth`), and other is unique for payment, so I left it here

Reply:
Pretty sure it ends up the same based on a quick audit, but I would also prefer the variable stays the same.